### PR TITLE
refactor(ucd-store): overhaul stores & implement tests (pt1)

### LINKED ISSUE #39: improve testing for `ucd-store` & `cli`
Currently our tests for the cli package, and ucd-store package is a bit vague.. And doesn't catch many of the edge cases i've run into.

### LINKED ISSUE #60: improve testing
This is a tracker issue for where we need to improve the testing

### LINKED ISSUE #62: [improve testing] handle edge cases of mirror

### LINKED ISSUE #63: [improve testing] handle edge cases of validate

### PR DESCRIPTION
**Pull request by luxass**
resolves #39 
resolves #60 
resolves #62 
resolves #63

# TODOS

## Store
- [ ] Complete the `clean` method (public)
- [ ] Complete the `analyze` method (public)
- [ ] Complete the `mirror` method (internal)
- [ ] Complete the `repair` method (public)
- [x] Refactor UCDStore to be based more on the FileSystemBridge
	- [x] Get rid of `local` & `remote`
	- [x] Utilize the `capabilities` property on bridges
- [ ] Improve Testing for UCD Store
- [x] Store Capabilities
> Based on the Bridge Capabilities determine if the store can use mirror, clean and so on.
- [x] Add decorator for methods to ensure correct capabilities.
- [ ] Add Type-Safe paths to Store
> Using a script to generate the paths, but still allow all strings.

## CLI

- [x] Add shared arguments to Store Commands
- [x] Handle UnsupportedCapabilitiesError in Commands

## Utils

- [x] Add capabilities object to Bridge
- [x] Add tests for bridge capabilities

## Summary by CodeRabbit

* **New Features**
  * Introduced unified UCD Store with dynamic capability detection and improved error handling.
  * Added new factory functions for creating UCD Store instances for Node.js, HTTP, and custom bridges.
  * Enhanced CLI commands for store analysis and cleaning, including version-specific cleaning and improved feedback.
  * Added in-memory and Node.js playgrounds for interactive testing.
  * Introduced custom error classes for more descriptive error reporting.
  * Added detailed logging utilities for playgrounds.
  * Improved HTTP and Node.js filesystem bridges with richer directory entry metadata and capability declarations.

* **Bug Fixes**
  * Improved handling of file system capabilities and error scenarios in both local and remote store modes.
  * Fixed directory listing and path handling inconsistencies in filesystem bridges.

* **Refactor**
  * Replaced legacy local/remote store logic with a single, capability-driven store class.
  * Removed deprecated file mirroring, validation, and repair utilities and related tests.
  * Updated file system bridge interfaces and directory listing output for consistency.
  * Updated CLI commands to use new store creation methods and improved error messaging.
  * Refined internal store initialization and manifest handling with schema validation.
  * Enhanced concurrency control utilities for store operations.
  * Streamlined error handling with custom error classes and capability assertions.
  * Updated TypeScript path mappings and enabled experimental decorators.

* **Tests**
  * Added comprehensive tests for store configuration, cleaning, capabilities, error handling, and utility functions.
  * Added new tests for concurrency utilities and error classes.
  * Updated and expanded test coverage for file system bridges and store behaviors.
  * Removed tests for deprecated mirroring and validation modules.

* **Chores**
  * Updated Node.js version requirements across all packages to >=22.13.
  * Enhanced TypeScript configurations and documentation settings.
  * Adjusted package exports and internal module paths for clarity and maintainability.
  * Improved HTTP mocking utilities with multi-method support and richer response helpers.

### CODE REVIEW

File: packages/utils/src/fs-bridge/node.ts [hunk 1]
```diff
@@ -21,16 +22,32 @@ async function safeExists(path: string): Promise<boolean> {
...
+    const entries = await readdir(path, {
       recursive: recursive ?? false,
+      withFileTypes: true,
     });
+
+    return entries.map((entry): FSEntry => ({
+      name: entry.name,
+      path: recursive ? `${entry.parentPath}/${entry.name}` : entry.name,
+      type: entry.isDirectory() ? "directory" : "file",
+    }));
```
[the remark this answers is not in the document]
Comment (luxass) [reply packages/utils/src/fs-bridge/node.ts:50]: `parentPath` is a real property on the `fs.Dirent` - it was added in `node v20.12`.

Can you check if we are using engines correctly for each package, and if not create an issue?

File: test/msw-utils/msw.ts [hunk 2]
```diff
@@ -10,61 +10,216 @@
...
+      }
+      return http[method](url, resolver);
+    });
+
+    MSW_SERVER.use(...handlers);
     return;
   }
 
   throw new Error("invalid arguments for mockFetch");
 }
```
[the remark this answers is not in the document]
Comment (luxass) [reply test/msw-utils/msw.ts:95]: I think this would make it worse, since the `${string}` can match everything.

**Comment by luxass** [comment]
Most of these changes is already merged into main, so closing for now.
